src/main/java/com/github/quiltservertools/ledger/mixin/blocks/FarmlandBlockMixin.java

```diff
@@ -20,25 +20,24 @@
 public abstract class FarmlandBlockMixin {
     @Inject(method = "onLandedUpon", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/FarmlandBlock;setToDirt(Lnet/minecraft/block/BlockState;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V"))
     public void logFarmlandTrampling(World world, BlockState state, BlockPos pos, Entity entity, float fallDistance, CallbackInfo ci) {
-        logFarmland(world, state, pos, entity);
+        logFarmland(world, state, pos, Sources.TRAMPLE, entity);
     }
 
     @Inject(method = "scheduledTick", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/FarmlandBlock;setToDirt(Lnet/minecraft/block/BlockState;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V"))
     public void logBlockAbove(BlockState state, ServerWorld world, BlockPos pos, Random random, CallbackInfo ci) {
-        logFarmland(world, state, pos, null);
+        logFarmland(world, state, pos, Sources.TRAMPLE, null);
     }
 
     @Inject(method = "randomTick", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/FarmlandBlock;setToDirt(Lnet/minecraft/block/BlockState;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V"))
     public void logRandomDecay(BlockState state, ServerWorld world, BlockPos pos, Random random, CallbackInfo ci) {
-        logFarmland(world, state, pos, null);
+        logFarmland(world, state, pos, Sources.DRY,null);
```

Comment:
Should be a space between the last two arguments. I'm inclined to stick with `trample` for all farmland to dirt conversions